components/locid_transform/src/fallback/algorithms.rs

```diff
@@ -457,7 +457,7 @@ mod tests {
             requires_data: true,
             extension_key: None,
             fallback_supplement: Some(LocaleFallbackSupplement::Collation),
-            expected_language_chain: &["yue-HK", "yue", "zh-Hant", "zh"],
+            expected_language_chain: &["yue-HK", "yue", "zh-Hant"],
```

Comment:
The only test failure: the `zh-Hant` -> `zh` collation fallback is removed in CLDR

Reply:
https://unicode.org/reports/tr35/#Parent_Locales says:

> The basic inheritance model for locales of the form lang_script_region_variant1_…variantN is to truncate from the end.
> ...
> In some cases, the normal truncation inheritance does not function well. For example, if the truncation algorithm changes script, then a mixture of child and parent textual data is a mishmash of different scripts.
> ...
>  there is a special attribute and value, localeRules="nonlikelyScript", which specifies all locales of the form lang_script, wherever the script is not the likely script for lang
> ...
> There are certain components that require addenda to the common parent fallback rules. For a locale like zh_Hant in the example above, the parentLocale element would dictate the parent as root when referring to main locale data, but for collation data, the parent locale should still be zh, even though the parentLocale element is present for that locale. To address this, components can have their own fallback rules that inherit from the common rules and add additional parents that supplement or override the common rules
> ...
> Note: When components were first introduced, the component-specific parent locales were be merged with the main parent locales. This was determined to be an error, and the component-specific parent locales are now not merged, but instead are treated as stand-alone.

So in other words, "nonlikelyScript" applies to only the base component, and the "collator" component needs full truncation fallback.

Concretely, it would be a big bug if `zh_Hant` started falling back to `root` for collation, because then users of `zh_Hant` would no longer get their collation tailorings.

Reply:
I think unfortunately the correct fix here might involve rewriting code, because we're not supposed to use _any_ of the common parent tables in the components any more.

Reply:
Can you write this code?

Reply:
Ehh... this code has to be re-written per #3867 and it's hard to make it backwards compatible. I want to fix #3867 but I don't want to block CLDR 45 on it.

Since we are already doing manual stuff here, can you just add the parent back in and cite #1964 and/or #3867 for a follow-up?